xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -248,6 +202,191 @@ private static ManagedChannel initLbChannel(
     return channel;
   }
 
+  /**
+   * A LoadBalancerProvider that provides a load balancer with a fixed XdsConfig.
+   */
+  private final class FixedXdsConfigBalancerProvider extends LoadBalancerProvider {
+    final XdsConfig xdsConfig;
+    final XdsConfig oldXdsConfig;
+    final EndpointWatcher oldEndpointWatcher;
+
+    FixedXdsConfigBalancerProvider(XdsConfig xdsConfig) {
+      this.xdsConfig = xdsConfig;
+      oldXdsConfig = LookasideLb.this.xdsConfig;
+      oldEndpointWatcher = endpointWatcher;
+    }
+
+    @Override
+    public boolean isAvailable() {
+      return true;
+    }
+
+    @Override
+    public int getPriority() {
+      return 5;
+    }
+
+    // A synthetic policy name identified by xds config.
+    @Override
+    public String getPolicyName() {
+      return "xds_policy__balancer_name_" + xdsConfig.balancerName
+          + "__childPolicy_" + xdsConfig.childPolicy
+          + "__fallbackPolicy_" + xdsConfig.fallbackPolicy
+          + "__edsServiceName_" + xdsConfig.edsServiceName
+          + "__lrsServerName_" + xdsConfig.lrsServerName;
+    }
+
+    @Override
+    public LoadBalancer newLoadBalancer(Helper helper) {
+      return new FixedXdsConfigBalancer(helper);
+    }
+
+    final class FixedXdsConfigBalancer extends LoadBalancer {
+      final Helper helper;
+
+      // All fields become non-null once handleResolvedAddresses() successfully.
+      // All fields are assigned at most once.
+      @Nullable
+      ObjectPool<XdsClient> xdsClientRef;
+      @Nullable
+      XdsClient xdsClient;
+      @Nullable
+      LocalityStore localityStore;
+      @Nullable
+      LoadReportClient lrsClient;
+
+      FixedXdsConfigBalancer(Helper helper) {
+        this.helper = helper;
+      }
+
+      @Override
+      public void handleNameResolutionError(Status error) {}
+
+      @Override
+      public boolean canHandleEmptyAddressListFromNameResolution() {
+        return true;
+      }
+
+      @Override
+      public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+        if (xdsClientRef != null) {
+          return;
+        }
+
+        LoadStatsStore loadStatsStore = new LoadStatsStoreImpl();
+        localityStore = localityStoreFactory.newLocalityStore(
+            helper, lbRegistry, loadStatsStore);
+        // TODO(zdapeng): Use XdsClient to do Lrs directly.
+        final LoadReportCallback lrsCallback =
+            new LoadReportCallback() {
+              @Override
+              public void onReportResponse(long reportIntervalNano) {
+                localityStore.updateOobMetricsReportInterval(reportIntervalNano);
+              }
+            };
+
+        if (xdsConfig.balancerName != null) {
+          // This to handle the legacy usecase that requires balancerName from xds config.
+
+          String oldBalancerName = oldXdsConfig == null ? null : oldXdsConfig.balancerName;
```

Comment:
Why you need this variable?

Reply:
If balancer name is changed, create a new channel/xdsClient, otherwise use the cached.

Reply:
I mean your code, you can directly use `oldXdsConfig.balancerName`, why you need to create another variable?

Reply:
I see what you mean. It just personal preference. I don't mind change to another way.